libp2p/muxers/mplex/mplex.nim

```diff
@@ -93,17 +97,20 @@ method handle*(m: Mplex) {.async, gcsafe.} =
                                    oid = m.oid
           if not isNil(m.streamHandler):
             let stream = newConnection(channel)
+            m.conns.add(stream)
             stream.peerInfo = m.connection.peerInfo
 
             var fut = newFuture[void]()
             proc handler() {.async.} =
-              tryAndWarn "mplex channel handler":
+              try:
                 await m.streamHandler(stream)
+              except CatchableError as exc:
```

Comment:
since handlers shouldn't raise, it should be a Defect if they do?

Reply:
yes, general rule is to not handle `Defect`, the consuming app can if it wants install a global exception handler to deal with it.